app/scripts/directives/deployImage.js

```diff
@@ -201,6 +205,62 @@ angular.module("openshiftConsole")
             });
             Navigate.toNextSteps($scope.app.name, $scope.project);
           };
+
+          var launchConfirmationDialog = function(alerts) {
+            var modalInstance = $uibModal.open({
+              animation: true,
+              templateUrl: 'views/modals/confirm.html',
+              controller: 'ConfirmModalController',
+              resolve: {
+                modalConfig: function() {
+                  return {
+                    alerts: alerts,
+                    message: "Problems were detected while checking your application configuration.",
+                    okButtonText: "Create Anyway",
+                    okButtonClass: "btn-danger",
+                    cancelButtonText: "Cancel"
+                  };
+                }
+              }
+            });
+
+            modalInstance.result.then(function() {
+              createResources();
+            });
+          };
+
+          var showWarningsOrCreate = function(result){
+            // Now that all checks are completed, show any warnings if we need to
+            var quotaWarnings = result.quotaWarnings || [];
+            var errorAlerts = _.filter(quotaWarnings, {type: 'error'}) || [];
+            if ($scope.nameTaken || errorAlerts.length) {
+              $scope.disableInputs = false;
+              $scope.alerts = quotaWarnings;
+            }
+            else if (quotaWarnings.length) {
+              launchConfirmationDialog(quotaWarnings);
+              $scope.disableInputs = false;
+            }
+            else {
+              createResources();
+            }
+          };
+
+          $scope.create = function() {
+            $scope.disableInputs = true;
+            $scope.alerts = {};
+            generatedResources = getResources();
+
+            var nameTakenPromise = ApplicationGenerator.ifResourcesDontExist(generatedResources, $scope.project);
+            var checkQuotaPromise = QuotaService.getLatestQuotaWarnings(generatedResources, $scope.context);
+            // Don't want to wait for the name checks to finish before making the calls to quota
+            // so kick off the requests above and then chain the promises here
+            var setNameTaken = function(result) {
+              $scope.nameTaken = result.nameTaken;
+              return checkQuotaPromise;
+            };
+            nameTakenPromise.then(setNameTaken, setNameTaken).then(showWarningsOrCreate, showWarningsOrCreate);
```

Comment:
i tried that originally, but `finally` does not get the values from resolve/reject returned to it